src/main/java/pl/put/poznan/buildingInfo/model/Building.java

```diff
@@ -70,4 +79,16 @@ public Float getLighting() {
         // return temporary value in order to compile code properly
         return (float) 0.0;
     }
+
+    public float getHeatingLimit() {
+        return heatingLimit;
+    }
+
+    public void setHeatingLimit(float heatingLimit) {
+        this.heatingLimit = heatingLimit;
+    }
+    public ArrayList<Room> getExceedingRooms() {
+        ArrayList<ArrayList<Room>> ar = (ArrayList) levels.stream().map(f -> f.getExceedingRooms(heatingLimit)).collect(Collectors.toList());
+        return (ArrayList) ar.stream().flatMap(List::stream).collect(Collectors.toList());
```

Comment:
Can't you do it one line?

Do we need to cast it on Array List?

Reply:
Couldn't we "flatten" that array? I'd like to test it and give a snippet we could discuss about 😉

Reply:
We should discuss, which one is better. I've played with oneline solution and it might be something like that `public ArrayList<Room> getExceedingRooms() {
        return (ArrayList) levels.stream().map(level -> level.getRooms().stream().filter(room -> (room.getHeating() / room.getCube()) > this.heatingLimit)
            .collect(Collectors.toList())).flatMap(l -> l.stream()).collect(Collectors.toList());
    }`

Reply:
If the solution you provided works I'd be happy to use it. I was about to get down to it and thought of a similar way of dealing with it.

Reply:
To be honest I would go for something like that:
```
(ArrayList) levels.stream()
            .map(f -> f.getExceedingRooms(heatingLimit))
            .flatMap(List::stream)
            .collect(Collectors.toList());
    }
```

Reply:
I've thought about solution not including method in Level class, but both are correct and solves our problem, so we shouldn't bother so much 😃